### PR TITLE
Additional DF Filters

### LINKED ISSUE #3097: Rearrange Data Files filters and implement multiselect fiscal period filter
**Description:**
On the data files page, implement multiselect functionality on STT filter 
Rearrange Filters for the Django Admin Data Files In the Following Order:
- STT
- Fiscal Period
- Year
- Quarter
- Section

**Acceptance Criteria:** 
- ~~[ ] Multiselect filter allows for filtering by fiscal period like on search index pages~~
- [ ] The filters are rearranged in the described order 
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
- [ ]  Implement multiselect for STT filter on Data Files page
- [ ] Rearrange order of filters
- [ ]  Confirm that the ability to select multiple STT locations passes by both groups
- [ ] Run Testing Checklist and confirm all tests pass

**Supporting Documentation:**
[- #2910 Mural Admin Exp Improvements](https://app.mural.co/t/raft2792/m/raft2792/1720460304271/b1de09b0efbfd35bb9f72063d4162ccd20a17d25?sender=70699c5a-0269-4ab3-8427-25b50f413642)

### LINKED ISSUE #3108: Additional datafile filters
**Description:**
To further specify which files need to be re-parsed, sys admins need more granular control.

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] Datafiles DAC page has granular options for narrowing/expanding datafile search per Notes
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
- [x] ~~Enable stacking/`OR`-ing of multiple filters of same type (e.g., I want all datafiles from Florida or Georgia in my query)~~ Implemented via multi-select #3183 
- [x] New Datafile filter for type (state, tribe, territory)
- [x] New Datafile filter for region -- likely same cross-link of datafile and STT objects as above
- [x] New Datafile filter for "latest" -- must be combined with other filter(s). We don't want to re-parse v1,v2,...,v35 of a file so the latest filter would be after the other filters return a queryset of datafiles, only cherry-pick the highest version of each.
- [x] New Datafile filter for program type (we currently have section (SSP Active))
- [x] New Datafile filter for section (piggybacks on above task)
- [x] Run Testing Checklist and confirm all tests pass

**Notes:**
- ~~Current filtering is "AND" logic which will provide a non-intuitive interface. The first task for stacking/`OR`-ing conditions is paramount to success.~~

- _Requested filters:
  - file id (e.g. 2053, lst)
  - STT (e.g. Alabama, lst)
  - STT type (e.g. tribe, lst)
  - region ( i.e. a set of STTs in a particular region)
  - program type (e.g. TANF)
  - most recent file version (e.g. latest)
  - processing status (e.g. pending)
  - section (e.g. Active Case Data)

**Supporting Documentation:**
- Please see MVP impl. via PR #3041 

**Open Questions:**

- how best to handle the scenario where parameters passed are in conflict (e.g. STT type = tribal and region = 1 (has no tribes)? 
  - Utilize confirmation dialog from #3106  
  - Can we safely rely on a downstream system to handle contradictory parameters as suggested? If using DRF filtering, we could just end up with 0 files and say nothing to do. 
  - Should we have an [Ansible-like](https://docs.ansible.com/ansible/2.9/user_guide/playbooks_checkmode.html) "preview" function? 
    - To be encoded as part of #3106

### PR DESCRIPTION
**Pull request by elipe17**
## Summary of Changes
- Added extra filters to filters list
Pull request closes #3108 and closes #3097

## How to Test
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.
2. Submit some files as different STTs across different regions
3. Filter as you please

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] New Datafile filter for type (state, tribe, territory)
- [x] New Datafile filter for region -- likely same cross-link of datafile and STT objects as above
- [x] New Datafile filter for "latest" -- implemented in #3139 
- [x] New Datafile filter for section (includes program type)
- [x] Testing Checklist has been run and all tests pass
- [x] filter rearranged according to #3097 
- [ ] README is updated, if necessary

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/data_files/admin/filters.py [hunk 1]
```diff
@@ -4,27 +4,6 @@
 from tdpservice.search_indexes.models.reparse_meta import ReparseMeta
 from tdpservice.core.filters import MostRecentVersionFilter
 
-class DataFileSummaryPrgTypeFilter(admin.SimpleListFilter):
```
Comment (elipe17) [inline tdrs-backend/tdpservice/data_files/admin/filters.py:7]: Per Alex:
Screenshot 2024-09-30 at 8 52 56 AM

File: tdrs-backend/tdpservice/data_files/admin/admin.py [hunk 2]
```diff
@@ -113,14 +113,16 @@ def queryset(self, request, queryset):
     ]
 
     list_filter = [
+        'id',
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/data_files/admin/admin.py]: won't this show *every* available `id`? at some point i imagine the list would get quite large. is there a point to having an `id` numeric filter like this? would a range or other filter mechanism be more useful?

**Review by jtimpe — APPROVED** [review]
Question about the `id` filter, but everything else looks good!

File: tdrs-backend/tdpservice/data_files/admin/admin.py (hunk 2 above)
Comment (elipe17) [reply tdrs-backend/tdpservice/data_files/admin/admin.py]: Alex confirmed it is unnecessary given what #3106 will provide.

**Review by raftmsohani — APPROVED** [review]
LGTM!

File: tdrs-backend/tdpservice/data_files/models.py [hunk 3]
```diff
@@ -164,8 +164,6 @@ def prog_type(self):
         # e.g., 'SSP Closed Case Data'
         if self.section.startswith('SSP'):
             return 'SSP'
-        elif self.section.startswith('Tribal'):
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/data_files/models.py:167]: @elipe17 just curious why we're dropping this. i didnt observe any weird behavior as a result but wondering if we removed this as part of dropping the program type filter?

Comment (elipe17) [reply tdrs-backend/tdpservice/data_files/models.py:167]: I removed it because it was an unnecessary branch since tribal and tanf both result in TANF

**Review by ADPennington — APPROVED** [review]
LGTM @elipe17 🚀 cc: @ttran-hub 

screenshot below shows the data file filters before (left) and after (right)

- program type removed
- stt type added
- region added 
- user added ❔ (not sure we needed this, but will not impact our work) 
- filtering re-arranged to spec
Screenshot 2024-10-10 141549_LI

**Pull request merged**